File: customcads/src/constants/data/cad.js
export default {
    name: {
        isRequired: true,
        maxLength: 18,
        minLength: 2,
    },
    description: {
        isRequired: true,
        maxLength: 750,
        minLength: 10,
    },
    categoryId: {
        isRequired: true,
    },
    price: {
        isRequired: true,
        min: 0.01,
        max: 1000
    },
    file: {
        isRequired: true
    },
    image: {
        isRequired: true
    }
};

File: CustomCADs.Domain/Entities/CAD.cs
using CustomCADs.Domain.Enums;
using CustomCADs.Domain.ValueObjects;

namespace CustomCADs.Domain.Entities;

public class Cad
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public required string Description { get; set; }
    public CadStatus Status { get; set; }
    public decimal Price { get; set; }
    public DateTime CreationDate { get; set; }
    public Coordinates CamCoordinates { get; set; } = new();
    public Coordinates PanCoordinates { get; set; } = new();
    public Paths Paths { get; set; } = new();
    public int CategoryId { get; set; }
    public Category Category { get; set; } = null!;
    public required string CreatorId { get; set; }
    public User Creator { get; set; } = null!;
}
